las_geoh5/export_las.py

```diff
@@ -0,0 +1,209 @@
+
+    :param file: lasio file object.
+    :param drillhole: geoh5py.drillhole object containing property
+        groups for collocated data.
+    :param group: Property group containing collocated float data
+        objects of 'drillhole'.
+    """
+
+    if group.depth_:  # type: ignore
+        file.append_curve("DEPTH", group.depth_.values, unit="m")  # type: ignore
+    else:
+        file.append_curve(
+            "DEPTH", group.from_.values, unit="m", descr="FROM"  # type: ignore
+        )
+        file.append_curve("TO", group.to_.values, unit="m", descr="TO")  # type: ignore
+
+    properties = [drillhole.get_data(k)[0] for k in group.properties]
+    for data in [
+        k for k in properties if k.name not in ["FROM", "TO", "DEPTH", "DEPT"]
+    ]:
+        file.append_curve(data.name, data.values)
+        if isinstance(data, ReferencedData):
+            for k, v in data.value_map.map.items():  # pylint: disable=invalid-name
+                file.params.append(
+                    HeaderItem(
+                        mnemonic=f"{data.name} ({k})", value=v, descr="REFERENCE"
+                    )
+                )
+
+    return file
+
+
+def add_survey_data(file: LASFile, drillhole: Drillhole | ConcatenatedDrillhole):
+    """
+    Add drillhole survey data to LASFile object.
+
+    :param file: las file object.
+    :param drillhole: drillhole containing survey data.
+
+    :return: Updated las file object.
+    """
+
+    # Add survey data
+    file.append_curve("DEPTH", drillhole.surveys[:, 0], unit="m")  # type: ignore
+    file.append_curve(
+        "DIP",
+        drillhole.surveys[:, 1],  # type: ignore
+        unit="degrees",
+        descr="from horizontal",
+    )
+    file.append_curve(
+        "AZIM",
+        drillhole.surveys[:, 2],  # type: ignore
+        unit="degrees",
+        descr="from north (clockwise)",
+    )
+
+    return file
+
+
+def write_curves(
+    drillhole: Drillhole | ConcatenatedDrillhole,
+    basepath: str | Path,
+    directory: bool = True,
+):
+    """
+    Write a formatted .las file with data from 'drillhole'.
+
+    :param drillhole: geoh5py drillhole object containing property
+        groups for collocated data.
+    :param basepath: Path to working directory.
+    :param directory: True if data is stored in sub-directories
+    """
+
+    if isinstance(basepath, str):
+        basepath = Path(basepath)
+
+    if not drillhole.property_groups:
+        raise AttributeError("Drillhole doesn't have any associated property groups.")
+
+    for group in drillhole.property_groups:
+        file = LASFile()
+        file = add_well_data(file, drillhole)
+        file = add_curve_data(file, drillhole, group)
+
+        if directory:
+            subpath = Path(basepath / group.name)
+            if not subpath.exists():
+                subpath.mkdir()
+        else:
+            subpath = basepath
+
+        survey_path = Path(subpath / f"{drillhole.name}.las")

```

Comment:
**suggestion (simpler):** `subpath` is a `Path`
```suggestion
        survey_path = subpath / f"{drillhole.name}.las"

```

Reply:
done.  Also found similar patterns and fixed.